enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,511 @@
+
+**Risk**: Afterburn changes could affect other CoreOS deployments
+**Mitigation**: Changes are isolated to KubeVirt provider detection and handling. Extensive testing in CI for both HyperShift and standalone CoreOS deployments.
+
+### Drawbacks
+
+- Adds complexity to the hypershift kubevirt controllers:
+  - implementing and ip pool mechanism
+  - Need to parse and understand the openshift network config format
+- Fix depend on a coreos change:
+  - Change should be accepted and merged by non hypershift related team.
+  - It will take some time for the RPM to reach RHCOS, also we will have to plan backports there.
+
+## Alternatives (Not Implemented)
+
+### Store all the ignition config directly at a secret instead of retrieve it using the ignition server
+
+Currently hypershift workers retrieve the ignition config from an ignition server, so they need to have network configured before that, an alternative
+would be to directly put the whole ignition config directly at the secret referenced by the VM so accessing ignition server is not needed.
+
+**Why not chosen**: The kubernetes resources has a size limit that has already being reache by hypershift when storing the whole ignition config there [Jira](https://issues.redhat.com//browse/OCPBUGS-60148), also it means changing how a hypershift fundamental part of the bootstrap process works.
+
+### Use the ovn-kubernetes localnet IPAM mechanism and implement the missing gateway and dns part
+
+With this solution IPAM configuration will be done at the ovn-kubernetes localnet user defined network instead of the hypershift kubevirt nodepool, but ovn-kubernetes is missing features for it:
+- There is no official support for localnet + IPAM
+- localnet is missing gateway and dns configuration
+
+Also this makes customer depend on ovn-kubernetes and localnet, while the solution on this enhancement makes it CNI independant, working even with other CNIs like cillium or calico.
+
+### Implement the ip pool at kubevirt ipam controller
+
+Instead of implementing the ip pool at cluster api provider kubevirt, there is one per hosted cluster, implement it at kubevirt ipam controller, there is one for all the hosted clusters,
+that would allow to overcome the ip overlapping issue between thosed clusters, although it's not real clear if it's a really a problem.
+
+Mayor problem is that there is no a clear way to specify what IPAM configuration to the controller since the VMI itself do not have enough "api" for it, and changing kubevirt API need
+clear justifications.
+
+### Add a DHCP server per network at hosted cluster control plane
+
+Implement a pod per network attaching something like a dnsmasq to that network and serve the configured IPAM.
+
+Main problem is the complexity of implementing proper storage backend for DHCP on a kubernetes environment, also
```

Comment:
And it may also conflict with the reasons why the customer refused to use DHCP in the first place